core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -395,7 +429,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
   private def mergeApplicationListing(fileStatus: FileStatus): Unit = {
     val newAttempts = try {
         val bus = new ReplayListenerBus()
-        val res = replay(fileStatus, bus)
+        val (res, count) = replay(fileStatus, bus)
+        metrics.eventReplayedCount.inc(count)
```

Comment:
You're using this metric in two different contexts: while creating the listing, and while building the UI.

That means there's no good way to get a useful metric out of it, since you don't know how many events correspond to each and you're measuring the times for each separately.